File: bug-tracker/src/main/java/rrpay/bug/repo/BugRepository.java
package rrpay.bug.repo;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;
import rrpay.bug.model.Bug;

import java.util.List;

@Repository
public interface BugRepository extends MongoRepository<Bug, String> {
    List<Bug> findByCode(String code);
    void deleteByCode(String code);
}

File: bug-tracker/src/main/java/rrpay/bug/kt/repo/BugRepository.kt
package rrpay.bug.kt.repo

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import rrpay.bug.kt.model.Bug

@Repository
interface BugRepository:MongoRepository<Bug,String> {
     findByCode(code:String): List<Bug>

     deleteByCode(code:String):Unit

}